valhalla/filesystem.h

```diff
@@ -491,4 +494,100 @@ inline std::chrono::time_point<std::chrono::system_clock> last_write_time(const
   return std::chrono::system_clock::from_time_t(FS_MTIME(s));
 }
 
+inline std::string generate_tmp_suffix() {
```

Comment:
all of the valhalla specific stuff in here should be removed and put into a different header. this header is supposed to be a drop in replacement for the functionality we would get from `std::filesystem` if we were on c++17, we shouldnt put valhalla specific logic here. sorry for the confusion